[XB1] Crashpad integration.

v-kryachko opened the pull request:
b/321046655

Change-Id: Ic3134a213a1b97db0a8ef5b82b94c181b9514c06

hlwarriner commented on starboard/BUILD.gn:
```diff
@@ -86,6 +86,9 @@ group("gn_all") {
     }
   }
+  if (sb_crashpad_integration) {
+    deps += [ "//third_party/crashpad/client" ]
```
Do you plan to use this interface directly in the xb1 port, instead of going through https://github.com/youtube/cobalt/blob/main/third_party/crashpad/wrapper/wrapper.h?

v-kryachko replied:
Yes, this time I included the client directly, as purpose of this wrapper is not clear to me yet. And the switch `sb_crashpad_integration` is rather a temporary solution.

hlwarriner reviewed (commented):
Can you please describe the scope of this PR? Is it just to get the Crashpad client interface buildable for xb1 - and then separate PRs will use the Crashpad client in the xb1 port, and get the  Crashpad handler executable building for xb1?

v-kryachko commented:
Updated the commits: removed change in file `starboard/win/shared/platform_configuration/BUILD.gn`
@andrewsavage1 Changes in //build are required to locate Visual Studio tools correctly when build with native toolchain.

andrewsavage1 commented:
> Updated the commits: removed change in file `starboard/win/shared/platform_configuration/BUILD.gn` @andrewsavage1 Changes in //build are required to locate Visual Studio tools correctly when build with native toolchain.

There are many changes to //build that look like they're just refactoring code, not changing anything functional in it. We want to keep changes to //build (which is upstream code) as minimal as possible, so please only make changes to it when it's absolutely necessary.

v-kryachko commented:
@andrewsavage1
Most of changes to //build can be moved to somewhere else, say, in //starboard/build? I'll try.

andrewsavage1 commented on build/config/win/BUILD.gn:
```diff
@@ -9,9 +9,6 @@ import("//build/config/compiler/compiler.gni")
 import("//build/config/sanitizers/sanitizers.gni")
 import("//build/config/win/control_flow_guard.gni")
 import("//build/config/win/visual_studio_version.gni")
-if (!is_starboard) {
```
Instead of trying to move these changes to //starboard/build, what you most likely want to try to do is switch these conditionals. For example, this could be switched to `if (!use_cobalt_customizations)`, as could the conditional on L5 of build/config/win/visual_studio_version.gni below.

v-kryachko replied:
Done.

andrewsavage1 commented on build/vs_toolchain.py:
Changes should not be made to this file. If we want to change the path, it should be done only in the GNI file build/config/win/visual_studio_version.gni

v-kryachko replied:
OK.

v-kryachko replied:
I moved most of changes from //build to //starboard/build for a while, before this code is moved in the right place.
